### ISSUE TITLE
How to login with passwordlesss

### ISSUE THREAD

**Issue by hughesxu** [paramiko/paramiko#1393]
I added the fix to support log with passwordless [https://github.com/mzakharo/paramiko/commit/6e979ea686975579a204472a1414c8d76c11d6eb]

And in order to establish connection, I use below codes:
`self.transport = paramiko.Transport((self.host, self.port))`
`self.transport.auth_none(username=self.username)`
`self.transport.connect(username=self.username, password=None)`
`self.client = paramiko.SSHClient() `
`self.client._transport = self.transport`

But I got "No existing session" failure, how to solve this?

**Comment by hughesxu** [comment]
Actually, paramiko.SSHClient objective have support passwordless itself, just tried:
```
self.client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
                self.client.connect(hostname=self.host, username=self.username, port=self.port, **password=None**, timeout=self.timeout, allow_agent=False, look_for_keys=False)
```
And I could login passwordless.

**Comment by MrMino** [comment]
Umm... are you sure? The call path in `SSHClient.connect()` never does an `auth_none()` anywhere, which would be required for a "passwordless" connection.

This was actually a big issue for me for the last 3 weeks, and I'm quite sure that `auth_none` is not used there.

**Comment by hughesxu** [comment]
> Umm... are you sure? The call path in `SSHClient.connect()` never does an `auth_none()` anywhere, which would be required for a "passwordless" connection.
> 
> This was actually a big issue for me for the last 3 weeks, and I'm quite sure that `auth_none` is not used there.
> 
> What server are you using?

Need to add the fix mzakharo/paramiko@6e979ea in code, then passwordless are supported
